invenio_app_ils/records/resolver/jsonresolver/related.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2018-2019 CERN.
+#
+# invenio-app-ils is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Resolve the referred related records."""
+
+import jsonresolver
+from invenio_pidstore.errors import PIDDoesNotExistError
+from werkzeug.routing import Rule
+
+from ...api import Document, IlsRecord, Series
+from ..resolver import get_field_value_for_record as get_field_value
+
+# Note: there must be only one resolver per file,
+# otherwise only the last one is registered
+
+
+@jsonresolver.hookimpl
+def jsonresolver_loader(url_map):
+    """Resolve the referred related records."""
+    from flask import current_app
+
+    def get_related_record(obj):
+        """Get related record based on object properties."""
+        return IlsRecord.get_record_by_pid(
+            obj["pid"],
+            pid_type=obj["pid_type"]
+        )
+
+    def related_resolver(record_class):
+        def resolver(pid):
+            """Resolve the referred related records."""
+            related = []
+            related_records = get_field_value(record_class, pid,
+                                              "related_records")
+            for obj in related_records:
+                record = get_related_record(obj)
+                delete_keys = ["related", "_computed"]
+                for key in delete_keys:
+                    if key in record:
+                        del record[key]
+                record["relation"] = obj["relation"]
+                related.append(record)
+            return related
+        return resolver
+
+    url_map.add(
+        Rule(
+            "/api/resolver/documents/<pid>/related",
```

Comment:
I could be wrong but is indexing not affected by this change? I mean if a document has a language relation to another one and the latter is updated the first shouldn't be reindexed? We have indexers to listen to another record type changes but not for the same record type...I think or I can be totally lost :P